source/core/options.ts

```diff
@@ -1428,6 +1429,10 @@ export default class Options {
 		}
 
 		if (url.hostname === 'unix') {
+			if (!this._internals.enableUnixSockets) {
+				throw new Error('Using UNIX domain sockets but option `enableUnixSockets` is not enabled');
+			}
+
```

Comment:
Followed @szmarczak's suggestion to check if the option is enabled. Throw an error if not enabled.